apps/src/music/views/MusicView.jsx

```diff
@@ -202,6 +200,10 @@ class UnconnectedMusicView extends React.Component {
       this.props.clearCallout();
     }
 
+    if (this.props.appName !== 'music') {
+      this.musicBlocklyWorkspace.hideFlyout();
+    }
```

Comment:
We want to hide the flyout after we've switched to a non-music level, to avoid a bug in which a highlighted flyout block would continue showing because the flyout has `visibility: visible` even when the music UI's root has `visibility: hidden` in that situation.

We could be more explicit about when do to this action if we were a functional React component and could handle the specific switch of app name from music to non-music.  Or we might even consider adding some additional lifecycle management calls into a Lab2 app, like we do for setup [here](https://github.com/code-dot-org/code-dot-org/blob/f57be23f85532f6607f93df95d68f9dd4be2626d/apps/src/lab2/views/LabViewsRenderer.tsx#L95).